src/AlgebraicGeometry/TropicalGeometry/groebner_fan.jl

```diff
@@ -0,0 +1,544 @@
+###
+# Computing Groebner fans in Oscar
+# ================================
+#
+# For a detailed writeup of the algorithm please see Anders Jensen's PhD dissertation
+#   "Algorithmic Aspects of Groebner Fans and Tropical Varieties"
+#
+# In tropical terminology: Groebner fans contain tropicalizations in the constant
+#   coefficient case. The convention is always max, as in OSCAR leading monomials
+#   under weighted orderings are the ones with maximal weighted degree.
+#
+###
+
+# Computes the space of weight vectors w.r.t. which G is weighted homogeneous
+@doc raw"""
+    homogeneity_space(G::Vector{<:MPolyElem})
+
+A `Cone` that is the space of all weight vectors under which all elements of `G` are weighted homogeneous.  The cone will be a linear subspace without any rays.
+
+# Note
+If `G` is a reduced Groebner basis of an ideal `I` with respect a global ordering, then the returned cone is the space of all weight vectors with respect to which `I` is weighted homogeneous.
```

Comment:
Please avoid lines which are too long (say < 100 chars, though as a rule of thumb I use between 78 and 92 lines). This helps with reviewing changes to these lines in GitHub and via `git` diff. It's just reality that if later we need to fix a typo in a line with for example 300 characters, it is a hassle to review this. 

Separately: maybe use an [admonition](https://docs.julialang.org/en/v1/stdlib/Markdown/#Admonitions) for notes.

```suggestion
!!! note
    If `G` is a reduced Groebner basis of an ideal `I` with respect a global
    ordering, then the returned cone is the space of all weight vectors with
    respect to which `I` is weighted homogeneous.
```

Reply:
Quick question: will the line breaks that I enter here manually cause line breaks in the documentation?